src/micro_http/src/server.rs

```diff
@@ -495,6 +496,9 @@ impl HttpServer {
 
     /// Changes the event type for a connection to either listen for incoming bytes
     /// or for when the stream is ready for writing.
+    ///
+    /// # Errors
+    /// `IOError` is returned when an `epoll::ctl` operation fails.
```

Comment:
Suggestion:
`IOError is returned when an epoll::ctl operation fails.` -> `IOError is returned when the modify operation on the events stream_fd is monitoring fails.`

Reply:
@lauralt , I don't get the last part of your suggestion, when you say that `is monitoring fails`. I think it is a good idea to mention details about the scenario when the error occurs, but keep it simple at the same time, and not imply parameters names, when the parameters list wasn't introduced in the documentation. As another suggestion, to stick along the scenario, we can say something like: 
```suggestion
    /// `IOError` is returned when an `EPOLL_CTL_MOD` control operation fails.
```
As well, I do not have a strong preference to mention the scenario when the error is thrown. It is fine either way, but with consistency.

Reply:
@lauralt as like @iulianbarbu suggest it will be more clear to the reader if we mention just action instead of explaining the process 🙂. WDYT?

Reply:
I'm fine either way too :+1:. Just noticed now that there are more functions in `server.rs` that have the same explanation in `#Errors` section, like `enqueue_responses()` and it's not super accurate, but we can come back to them in a future PR.